### DISCUSSION TITLE
HTML Autoclose Tags

### DISCUSSION THREAD

**Discussion by jgttech** [zed-industries/zed#6592] [Feature Requests]
Is there a mechanism that can be configured (or plugin) that can be leveraged to auto-close HTML-style tags (i.e JSX, etc)???

**Comment by fabricio-fn** [comment]
use this extension: Auto Rename Tag

![Screenshot_2023-08-14_12-06-11](https://github.com/zed-industries/community/assets/116690430/788a34e1-9bd2-42e2-88c7-96a58230a2fb)

**Reply by teziovsky** [reply]
How to use it in Zed?

**Reply by fabricio-fn** [reply]
and a vs code extension, I don't know if it's on zed

**Reply by JosephTLyons** [reply]
We don't have a plugin system yet, but we will in the future.

**Reply by teziovsky** [reply]
@JosephTLyons is the „future” described by date? For example in which month or year? Do you have any timeline for new features that you plan to implement? 🤔

**Reply by JosephTLyons** [reply]
The plugin system release is very vague.  We've jumped into working on it a few times, but had to push it over for various reasons each time. Right now, we are sprinting to build the features we want Zed to have so we can start allowing the community in to our codebase. I think that we will likely focus on language extensibility and plugins once our codebase becomes public. Unfortunately, we don't have a public roadmap, though to be entirely honest, we don't have deadlines nailed down ourselves, things are sort of vague to us as well, as we are building things out.

**Comment by JosephTLyons** [comment]
@jgttech - at the moment, this doesn't exist in Zed.  Currently, I'm not sure if we will bake it in or let it become a plugin once plugin support lands.

**Comment by danielR1234** [comment]
@JosephTLyons 

I believe this feature should be given the highest priority. It's practically impossible to write any JSX code or similar without it. Like many others, I work a lot with React, and it's essentially a must-have. That's the main reason why I haven't switched to your IDE.

**Reply by tahalakrati** [reply]
Absolutely agree!

**Reply by glenmiracle18** [reply]
they should 😩

**Reply by sidvud98** [reply]
💯💯

**Reply by pedroSoaresll** [reply]
yes, I am trying Zed these days, but without this feature, it is complicated to proceed. I hope this feature can come soon 🙏🏽

**Reply by michaelajayi** [reply]
I feel the same here, it's just at this point my tradeoff for speed (whisper: vs code can be so annoying with performance these days, it's bloated!), I'll stick around till I or someone release a plugin for this issue out of abundant daily frustration. Win-win you'd say.

**Comment by DziugisLT** [comment]
I think this should be baked in as an option in the settings (or somewhere else) so it could be turned on/off (for anyone who would like that) because it is so widely used.

**Comment by alea12** [comment]
FYI this feature is tracked in this issue as well. Adding reactions might help drawing attention of the dev team: https://github.com/zed-industries/zed/issues/5267

**Comment by JosephTLyons** [comment]
The HTML extension now supports autoclosing of tags:

- https://github.com/zed-industries/zed/issues/5267
